.github/workflows/inspektor-gadget.yml

```diff
@@ -359,67 +359,92 @@ jobs:
+      uses: ./.github/actions/run-integration-tests
+      env:
+        KUBERNETES_DISTRIBUTION: "aro"
+      with:
+        registry: ${{ secrets.CONTAINER_REGISTRY }}
+        username: ${{ secrets.CONTAINER_REGISTRY_USERNAME }}
+        password: ${{ secrets.CONTAINER_REGISTRY_PASSWORD }}
+
   test-integration:
     name: Integration tests
     # level: 1
     needs: [test-unit, build-kubectl-gadget, build-local-gadget, build-docker-image]
     runs-on: ubuntu-latest
     steps:
+    - name: Check out code
+      uses: actions/checkout@v1
     - name: Setup Minikube
       uses: manusa/actions-setup-minikube@v2.4.3
       with:
         minikube version: 'v1.25.2'
         kubernetes version: 'v1.23.0'
         github token: ${{ secrets.GITHUB_TOKEN }}
-    - name: Set IMAGE_TAG
-      run: |
-        TMP1=${GITHUB_REF#*/}
-        TMP2=${TMP1#*/}
-        IMAGE_TAG=${TMP2//\//-}
-        if [ "$IMAGE_TAG" = "main" ]; then
-            IMAGE_TAG="latest"
-        fi
-        echo IMAGE_TAG=$IMAGE_TAG >> $GITHUB_ENV
-    - name: Check out code
-      uses: actions/checkout@v1
-    - name: Get kubectl-gadget-linux-amd64.tar.gz from artifact.
-      uses: actions/download-artifact@v2
-      with:
-        name: kubectl-gadget-linux-amd64-tar-gz
-        path: /home/runner/work/inspektor-gadget/
-    - name: Integration tests
-      run: |
-        echo "Using IMAGE_TAG=$IMAGE_TAG"
-
-        tar zxvf /home/runner/work/inspektor-gadget/kubectl-gadget-linux-amd64.tar.gz
-        mv kubectl-gadget kubectl-gadget-linux-amd64
-
-        TESTS_DOCKER_ARGS="-e KUBECONFIG=/root/.kube/config -v /home/runner/.kube:/root/.kube -v /home/runner/work/_temp/.minikube:/home/runner/work/_temp/.minikube" \
-            make -C integration build test
-
-        sed -i "s/latest/$IMAGE_TAG/g" integration/gadget-integration-tests-job.yaml
-    - name: Add integration asset as artifact.
-      if: startsWith(github.ref, 'refs/tags/v')
-      uses: actions/upload-artifact@master
-      with:
-        name: integration-asset
-        path: /home/runner/work/inspektor-gadget/inspektor-gadget/integration/gadget-integration-tests-job.yaml
-    - name: Login to Container Registry
-      if: startsWith(github.ref, 'refs/tags/v')
-      uses: docker/login-action@v1
+    - name: Run integration tests
+      uses: ./.github/actions/run-integration-tests
+      env:
+        KUBERNETES_DISTRIBUTION: "minikube-github"
       with:
         registry: ${{ secrets.CONTAINER_REGISTRY }}
         username: ${{ secrets.CONTAINER_REGISTRY_USERNAME }}
         password: ${{ secrets.CONTAINER_REGISTRY_PASSWORD }}
-    - name: Push Integration Test Image
-      if: startsWith(github.ref, 'refs/tags/v')
-      run: |
-        make -C integration push
 
   release:
     name: Release
     # level: 2
-    needs: [documentation-checks, lint, test-integration, test-local-gadget]
+    needs: [documentation-checks, lint, test-integration, test-integration-aro, test-local-gadget]
```

Comment:
Do you agree to make `test-integration-aro` a need of the `release` job? Take into account that `test-integration-aro` is now optional (based on secrets' existence) thus if we do not configure an ARO cluster on the repo secrets both `test-integration-aro` and `release` will be skipped.

Reply:
I am not sure to understand, I would imagine if a job is skipped then it fufills needs condition.

Reply:
No, if the `test-integration-aro` job is skipped, all the jobs that have it as a `needs` condition will be skipped too. However, in this repo, we will always have the ARO secrets configured so it should never be skipped.